src/containers/GraphSettingsContainer.tsx

```diff
@@ -55,6 +58,14 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps> {
     router: PropTypes.object
   };
 
+  constructor(props: GraphSettingsProps) {
+    super(props);
+    // ensure setting is initialized to match the url
+    if (props.showServiceNodes !== props.injectServiceNodes) {
```

Comment:
This `showServiceNode` vs `injectServiceNode` is confusing to me

Reply:
Ah yes. The url should drive the page. So any url changes should also update the Redux state as well so everything is in sync.

Reply:
> This showServiceNode vs injectServiceNode is confusing to me. Could we just use one?

Do you mean: could we combine them into one setting?  I guess the answer is that we probably could.  But at the moment all of the URL props are held in session state.  And all of the graph settings are held in redux.  By keeping the different names it helped me remember which was which.  I agree that it seems it should be a single setting but maybe it can be done when we move everything to redux.  At least for now the URL controls things.

If you have a specific recommendation for how to do it let me know and we can try it.

Reply:
we can take care of that in the other Redux PR; because redux will eliminate session state

Reply:
Our goal is to abolish all session state in favor of Redux. There can only be one! ;)

Reply:
And having *one* with tools and plugins is way cooler.